### PR TITLE
fixes #32179 - Return code 404 instead of 500 missing subnet

### PR DESCRIPTION
**Pull request by domitea**
Bootdisk should return code 404 for missing subnet

### CODE REVIEW

File: app/controllers/foreman_bootdisk/api/v2/subnet_disks_controller.rb [hunk 1]
```diff
@@ -42,6 +43,10 @@ def subnet
         def resource_scope
           Subnet.authorized('view_subnets')
         end
+
+        def subnet_not_found
+          not_found ("Subnet not found by id '%s'") % params[:id]
```
Comment (lzap) [inline app/controllers/foreman_bootdisk/api/v2/subnet_disks_controller.rb:48]: Subnet with id %s not found

sounds probably little bit better. Tests fail tho.

**Comment by lzap** [comment]
There are (irrelevant) rubocop failures, can you fix them please?

File: lib/foreman_bootdisk/engine.rb [hunk 2]
```diff
@@ -57,6 +57,11 @@ class Engine < ::Rails::Engine
         provision_method 'bootdisk', N_('Boot disk based')
         template_labels 'Bootdisk' => N_('Boot disk embedded template')
         allowed_template_helpers :bootdisk_chain_url, :bootdisk_raise
+
+        extend_page "subnets/index" do |cx|
+          cx.add_pagelet :modals, partial: 'subnets/bootdisk_modal'
+          cx.add_pagelet :title_action_buttons, name: 'Bootdisk', partial: 'subnets/index_bootdisk_button', priority: 10000
+        end
       end
     end
```
Comment (lzap) [inline lib/foreman_bootdisk/engine.rb:67]: These three files do not seem to be related, was this added by mistake?

Comment (domitea) [reply lib/foreman_bootdisk/engine.rb]: Yes, the mistake - that's happen when you're working on the bootdisk improvements at many fronts :/

File: app/controllers/foreman_bootdisk/api/v2/subnet_disks_controller.rb (hunk 1 above)
Comment (lzap) [reply app/controllers/foreman_bootdisk/api/v2/subnet_disks_controller.rb:48]: I'd still prefer a cleaner message, but whatever.

**Review by lzap — APPROVED** [review]
I am good, thanks.
